src/utils/get-additional-sources.ts

```diff
@@ -0,0 +1,319 @@
+import buildUrl from 'build-url'
+
+import type { MediaType } from '~/constants/media'
+import type { ApiQueryParams } from '~/store/types'
+
+import type { BuildUrlOptions } from 'build-url'
+
+/**
+ * Describes the query format used by the URL builder functions of additional
+ * sources. This includes the query string and the usage filters.
+ */
+interface AdditionalSearchQuery {
+  q: string
+  filters: {
+    commercial: boolean
+    modify: boolean
+  }
+}
+
+/**
+ * Convert the ApiQueryParams object from the store to a format used by all the
+ * URL builder functions of all additional sources.
+ *
+ * @param query - the ApiQueryParams object
+ * @returns the query and filters in the format used by the URL builders
+ */
+const transformSearchQuery = (
+  query: ApiQueryParams
+): AdditionalSearchQuery => ({
+  q: query.q ?? '',
+  filters: {
+    commercial: query.license_type?.includes('commercial') ?? false,
+    modify: query.license_type?.includes('modification') ?? false,
+  },
+})
+
+type SearchFunctions = {
+  [k in MediaType]?: (
+    search: AdditionalSearchQuery
+  ) => BuildUrlOptions & { url: string }
+}
+
+/**
+ * , which contains a display name, whether the
```

Comment:
Something is missing in the docstring

Reply:
I'm confused as to how that bit got deleted! Thanks for catching this.